futures-util/src/compat/compat03as01.rs

```diff
@@ -128,32 +115,16 @@ impl Current {
         Current(task01::current())
     }
 
-    fn as_waker(&self) -> LocalWakerLt {
-        unsafe {
-            LocalWakerLt {
-                inner: task03::LocalWaker::new(NonNull::new_unchecked(self as *const Current as *mut Current)),
-                _marker: PhantomData,
-            }
-        }
-    }
-}
-
-unsafe impl task03::UnsafeWake for Current {
-    #[inline]
-    unsafe fn clone_raw(&self) -> task03::Waker {
-        task03::Waker::from(Arc::new(Current(self.0.clone())))
-    }
-
-    #[inline]
-    unsafe fn drop_raw(&self) {} // Does nothing
-
-    #[inline]
-    unsafe fn wake(&self) {
-        self.0.notify();
+    fn as_waker(&self) -> Waker {
+        // For simplicity reasons wrap the Waker into an Arc.
+        // We can optmize this again later on and reintroduce WakerLt<'a> which
+        // derefs to Waker, and where cloning it through RawWakerVTable returns
+        // an Arc version
```

Comment:
:+1:

Reply:
I think you meant "optimize" here and not "optmize?"

Reply:
This is now cloning the current task twice on every single poll.

Reply:
Yeah, we'll want to follow up with this optimization pretty quickly since this will cause an allocation every time `as_waker` is called :(